synapse/storage/persist_events.py

```diff
@@ -735,9 +736,32 @@ def _maybe_start_persisting(self, room_id: str):
                 if event_id == event.event_id:
                     if event.membership == Membership.JOIN:
                         return True
+                else:
+                    events_to_check.append(event_id)
```

Comment:
this should surely only happen if the event is nowhere to be found in `ev_ctx_rm` (ie, if we hit the end of the `for` loop over `ev_ctx_rm`)